Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -3,17 +3,68 @@ import Foundation
 /// `BSONDecoder` facilitates the decoding of BSON into semantic `Decodable` types.
 public class BSONDecoder {
 
+    @available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
```

Comment:
If I understand the `@available` docs on [Attributes](https://docs.swift.org/swift-book/ReferenceManual/Attributes.html) correctly, this means that `iso8601Formatter` is available on all Swift versions but only the above platforms. If so, how does this interact with Linux builds? Do those only use the Swift language version requirement (i.e. always available in this case)?

Side note: I assume these requirements were pulled from the column in [`date(from:)` docs](https://developer.apple.com/documentation/foundation/iso8601dateformatter/1643127-date) (or a similar page).

Reply:
Yeah, it appears to be platform independent on Linux (The formatter exists going back to 4.0 https://github.com/apple/swift-corelibs-foundation/blob/swift-4.0-branch/Foundation/ISO8601DateFormatter.swift). I honestly don't understand why it has platform specific availability instead of Swift version on Apple devices, but it seems to be the case based on the documentation you mentioned and the code in `JSONEncoder`.

I tested it manually on Ubuntu 18.04 and Swift 4.2, and the formatter was available as expected.